common/src/jni/main/cpp/NativeCrypto.cpp

```diff
@@ -4414,6 +4437,38 @@ static jlong NativeCrypto_asn1_read_uint64(JNIEnv* env, jclass, jlong cbsRef) {
     return value;
 }
 
+static void NativeCrypto_asn1_read_null(JNIEnv* env, jclass, jlong cbsRef) {
+    CbsHandle* cbs = reinterpret_cast<CbsHandle*>(static_cast<uintptr_t>(cbsRef));
+    JNI_TRACE("asn1_read_null(%p)", cbs);
+
+    CBS null_holder;
+    if (!CBS_get_asn1(cbs->cbs.get(), &null_holder, CBS_ASN1_NULL)) {
+        Errors::throwIOException(env, "Error reading ASN.1 encoding");
+    }
+}
+
+static jstring NativeCrypto_asn1_read_oid(JNIEnv* env, jclass, jlong cbsRef) {
+    CbsHandle* cbs = reinterpret_cast<CbsHandle*>(static_cast<uintptr_t>(cbsRef));
+    JNI_TRACE("asn1_read_oid(%p)", cbs);
+
+    CBS oid_cbs;
+    if (!CBS_get_asn1(cbs->cbs.get(), &oid_cbs, CBS_ASN1_OBJECT)) {
+        Errors::throwIOException(env, "Error reading ASN.1 encoding");
```

Comment:
It would probably make sense to have different exception messages for the IOExceptions thrown on line 4461 and 4466 so that it's easier to figure out where things went wrong..